lib/util/got/got.spec.ts

```diff
@@ -1,6 +1,7 @@
 import nock from 'nock';
 import { getConfigResponse } from '../../datasource/docker';
 
+// TODO: move to datasource/docker ?
```

Comment:
Yes, I think so

Reply:
Maybe we should move the docker auth logic (beforeRedirect) to central got, because the url based auth like `X-Amz-Algorithm` can be used by more datasource in future. We never know if some registry will move there. So it will simply work for all datasources.

Reply:
I will leave it here for now and refactor it in a later pr.